src/Swift.Bindings/src/Emitter/StringEmitter/Handler/TypeHandler.cs

```diff
@@ -74,19 +74,28 @@ public void Emit(CSharpWriter csWriter, SwiftWriter swiftWriter, IEnvironment en
             csWriter.WriteLine($"public unsafe struct {structDecl.Name} : {typeof(ISwiftObject).Name} {{");
             csWriter.Indent++;
 
-            // Emit each field in the struct
-            foreach (var fieldDecl in structDecl.Fields)
+            // For frozen structs, we need to emit fields that match the Swift struct's memory layout exactly.
+            // These backing fields are required for proper memory layout and marshalling, even though they
+            // are never directly accessed from C# code. The actual value access happens through Swift's
+            // accessor methods.
+            //
+            // Important: Direct access to these fields from C# will not provide the correct value - always
```

Comment:
I would write this into the generated code. One useful property of generating C# is that others can look at it/debug it. Even for projections, if you Go To Definition in VS/VS Code it will use source link on the NuGet and download the source code from github.